### PR TITLE
kmplayer Cheat Sheet

### PR DESCRIPTION
**Pull request by mesooryanarayananb**
**What does your Instant Answer do?**

Give an instant answer to a searched Keyword

**What problem does your Instant Answer solve (Why is it better than organic links)?**

It gives cheatsheets for KMplayer 

**What is the data source for your Instant Answer? (Provide a link if possible)**

KMplayer Software , The data source is the software , I reviewed each menus in it.

**Why did you choose this data source?**

Because KMplayer is an widely used Software . I hope it will be helpful for many users

**Are there any other alternative (better) data sources?**

I dont no , lets duck it !!!

**What are some example queries that trigger this Instant Answer?**
Kmplayer , KMplayer cheatsheet

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
People who use KMplayer to see movies and videos

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No ,I think

**Which existing Instant Answers will this one supercede/overlap with?**
none

**Are you having any problems? Do you need our help with anything?**
I want to start tech events about duckduckgo in my college , so maybe some interested students will create cheatsheets and instantanswer for duckduckgo . So can I do that ? If yes ! Is there any procedures ??

**Where did you hear about DuckDuckHack? (For first time contributors)**
Hackton Event Kochi

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [x] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```
screenshot 41

-----
IA Page: https://duck.co/ia/view/kmplayer_cheat_sheet

### CODE REVIEW

**Comment by gautamkrishnar** [comment]
@mesooryanarayananb Code looks good :smile:

**Comment by gautamkrishnar** [comment]
# all checks have passed
@moollaza its a first contribution from @mesooryanarayananb 
Please guide him if you need any changes..

**Comment by moollaza** [comment]
@mesooryanarayananb thanks a lot for contributing, we really appreciate it!

@gautamkrishnar thanks for reviewing :+1:

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+                "val": "Delete a file playing"
+            },
+            {
+                "key": "[shift] [Ctrl] [Del] ",
+                "val": "Delete a file playing instantly"
+            }
+        ],
+        "Option": [
+            {
+                "key": "[F2]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: For any single keys, without spaces you shouldn't have any `[]`. They're only used when your `key` has spaces ` ` in it, to specify which parts need to be wrapped in `<code>` blocks to specify a keyboard key.

If you look at the output from your Cheat Sheet there are several keys which have `[]` around them (like this one). Let's make sure nothing is showing a `[` or `]` unless you want it to :)

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "[Ctrl] [F1]",
+                "val": "Motion Blur"
+            },
+            {
+                "key": " [Ctrl] [F2]",
+                "val": "Soften"
+            },
+            {
+                "key": " [Ctrl] [F3]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Please remove and leading/trailing spaces

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "[Alt] [F2]",
+                "val": "Font size : Decrease"
+            },
+            {
+                "key": "[Alt] [F3]",
+                "val": "Size Reset"
+            }
+        ],
+        "PLayback": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: typo: "Playback"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 4]
```diff
@@ -0,0 +1,394 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "Kmplayer",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Should this say "K-Multimedia Player"? I'm not sure which name is more popular?

Perhaps "K-Multimedia Player (KMPlayer) is best?

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+    "description": "Multimedia",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Let's remove this as it doesn't really explain too much.

File: share/goodie/cheat_sheets/json/kmplayer.json (hunk 4 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Seeing as there are different names, let's add add an `aliases` property:
```json
"aliases": [
    "the kmplayer",
    "k-multimedia player",
    "..."
]
```

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 6]
```diff
@@ -0,0 +1,394 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "Kmplayer",
+    "description": "Multimedia", 
+    "metadata": { 
+        "sourceName": "KmplayerCheatSheet"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: We need both a `sourceName` and `sourceURL` if you want to provide a `More at` link. What source did you use to find all these shortcuts? Please provide it's name and url so users can visit the source to learn more.

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "[Shift] [F9]",
+                "val": "Crystality"
+            },
+            {
+                "key": "[Shift] [Bksp]",
+                "val": "Ignore All Settings "
+            }
+        ],
+        "Video(Advanced )": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Typo: trailing space after `Advanced`

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "[Alt] [F]",
+                "val": "Super Speed Mode"
+            }, 
+            {
+                "key": "[Alt] [Ctrl] [F]",
+                "val": "High Speed Mode"
+            }
+        ],
+        "Video(basic)": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Let's put a space between before the `(`. "Video (basic)"

File: share/goodie/cheat_sheets/json/kmplayer.json (hunk 7 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Let's add a space before the `(` here as a well

**Comment by moollaza** [comment]
@mesooryanarayananb I've left some feedback. Please drop a comment here once you've updated everything :+1:

**Comment by mesooryanarayananb** [comment]
@moollaza is this look ok :smile:

**Comment by gautamkrishnar** [comment]
:smile:  LGTM :+1:

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 9]
```diff
@@ -0,0 +1,398 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "K-Multimedia Player (KMPlayer)",
+    "aliases": [
+    "the kmplayer",
+    "k-multimedia player"
+    ]
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Whoops, you need a comma here: `],`

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 10]
```diff
@@ -0,0 +1,398 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "K-Multimedia Player (KMPlayer)",
+    "aliases": [
+    "the kmplayer",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Make sure these are properly indented as well, please

**Comment by mesooryanarayananb** [comment]
@moollaza check it out

**Comment by mesooryanarayananb** [comment]
@moollaza removed [] for singlekeys , is everythng okay now ????

**Comment by gautamkrishnar** [comment]
:+1:

**Comment by mesooryanarayananb** [comment]
@gautamkrishnar evrythng oky ??

**Comment by gautamkrishnar** [comment]
@mesooryanarayananb LGTM :+1: 
but @moollaza will help you out if he need more changes...

**Comment by mesooryanarayananb** [comment]
@gautamkrishnar  okay :)

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+                "key": " [Ctrl] [F9]",
+                "val": "Auto Level Control"
+            },
+            {
+                "key": " [Ctrl] [F10]",
+                "val": "Miror Image"
+            },
+            {
+                "key": " [Ctrl] [F11]",
+                "val": "filp input picture"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Typo with "flip"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+                "key": " [Ctrl] [F8]",
+                "val": "Grayscale"
+            },
+            {
+                "key": " [Ctrl] [F9]",
+                "val": "Auto Level Control"
+            },
+            {
+                "key": " [Ctrl] [F10]",
+                "val": "Miror Image"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Typo with "Mirror"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 13]
```diff
@@ -0,0 +1,394 @@
...
+   "template_type": "keyboard",
+    "section_order": [  
+        "Open",
+        "Album",
+        "Screen Controls",
+        "Pan & Scan ",
+        "PLayback",
+        "Subtitles",
+        "Video(basic)",
+        "Video(Advanced )",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Remove space after "Advanced"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 14]
```diff
@@ -0,0 +1,394 @@
...
+    "description": "Multimedia", 
+    "metadata": { 
+        "sourceName": "KmplayerCheatSheet"
+    },
+   "template_type": "keyboard",
+    "section_order": [  
+        "Open",
+        "Album",
+        "Screen Controls",
+        "Pan & Scan ",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Remove space after "Scan"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 15]
```diff
@@ -0,0 +1,394 @@
...
+        "PLayback",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Lower case "L" in "Playback"

File: share/goodie/cheat_sheets/json/kmplayer.json (hunk 13 above)
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Oh and add a space as well like "Video (basic)" and "Video (advanced)"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 16]
```diff
@@ -0,0 +1,394 @@
...
+                "key": "[Shift] [F6]",
+                "val": "Remove Right Channel"
+            },
+            {
+                "key": "[Shift] [F7]",
+                "val": "Swap Channels"
+            }, 
+            {
+                "key": "[Shift] [F8]",
+                "val": "Downmi to Mono"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Looks like there's a typo here

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 17]
```diff
@@ -0,0 +1,394 @@
...
+                "key": "[Alt] [W]",
+                "val": "aWarpSharp"
+            }, 
+            {
+                "key": "[Alt] [B]",
+                "val": "Soften (Blur)"
+            }, 
+            {
+                "key": "[Alt] [N]",
+                "val": "Gradual Denase"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Should it be "Gradual Denoise"?

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 18]
```diff
@@ -0,0 +1,394 @@
...
+                "key": "[Alt] [X]",
+                "val": "Show/Hide Subtitles"
+            },
+            {
+                "key": "[Alt] [F1]",
+                "val": "Font size : Increase"
+            },
+            {
+                "key": "[Alt] [F2]",
+                "val": "Font size : Decrease"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Better to reword as "Increase font size" and "Decrease font size"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 19]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "F",
+                "val": "Frame Step"
+            },
+            {
+                "key": "[Shift] [F]",
+                "val": "Frame Step Back"
+            }, 
+            {
+                "key": "Pgup",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Style as "Page Up"

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 20]
```diff
@@ -0,0 +1,394 @@
...
+                "val": "Half Size"
+            },
+            {
+                "key": "[Alt] [2]",
+                "val": "Normal Size"
+            },
+         
+            {
+                "key": "[Alt] [3]",
+                "val": "One & Half size"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Maybe better as "One half size"?

**Comment by mesooryanarayananb** [comment]
@jagtalon  all okay now ???

File: share/goodie/cheat_sheets/json/kmplayer.json (hunk 2 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: @mesooryanarayananb this still needs to be addressed

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 21]
```diff
@@ -0,0 +1,394 @@
...
+                "val": "Flip input picture"
+            },
+            {
+                "key": " [Ctrl] [F12]",
+                "val": "Flip Output Picture"
+            }
+        ],
+        "Subtitles": [
+            {
+                "key": "[Ctrl ][Alt] [Q]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: We need a space between `[Ctrl]` and `[Alt]`

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 22]
```diff
@@ -0,0 +1,394 @@
...
+            {
+                "key": "[Alt] [4]",
+                "val": "Double Size"
+            },
+            {
+                "key": "[Alt] [Enter]",
+                "val": "Full Screen (Normal) "
+            },
+            {
+                "key": "[Ctrl ][Alt] [Enter]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Here too, need a space between the two keys (and the space in `[Ctrl]` should be removed

**Comment by moollaza** [comment]
@mesooryanarayananb thanks for the fixes, there's just a few things left and then this will be good to merged:

- some typos and leading/trailing spaces in the `keys`
- It seems like the most "proper" format for the name is "K-Multimedia Player"? https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1510#discussion_r38565370
- We still need to add some aliases: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1510#discussion_r38565705

**Comment by mesooryanarayananb** [comment]
@moollaza  , all good ?? changed the name to K-multimedia player ! and fixed the typos..

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 23]
```diff
@@ -0,0 +1,399 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "K-Multimedia Player",
+    "description": "Multimedia Player Cheat Sheet", 
+    "metadata": { 
+        "sourceName": "KmplayerCheatSheet"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Whoops, looks like you removed the sourceUrl? You need the URL if you're going to provide a source.

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 24]
```diff
@@ -0,0 +1,399 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "K-Multimedia Player",
+    "description": "Multimedia Player Cheat Sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Let's remove "Cheat Sheet" from the description.

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 25]
```diff
@@ -0,0 +1,399 @@
+{
+    "id": "kmplayer_cheat_sheet", 
+    "name": "K-Multimedia Player",
+    "description": "Multimedia Player Cheat Sheet", 
+    "metadata": { 
+        "sourceName": "KmplayerCheatSheet"
+    },
+    
+    "aliases": [
+        "kmplayer"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: these should be different from the filename. Seeing as the filename is `kmplayer.json` the query `kmplayer cheat sheet` will trigger this. If we wants `k-multimedia player cheat sheet` to trigger this though, we need to `k-multimedia player` as an `alias`:

```json
"aliases" : [ "k-multimedia player", "other name for kmplayer", "..." ],
```

File: share/goodie/cheat_sheets/json/kmplayer.json [hunk 26]
```diff
@@ -0,0 +1,399 @@
...
+                "val": "Exit Kmplayer"
+            }
+        ],
+        "Playlist": [
+            {
+                "key": "[Alt] [E]",
+                "val": "Playlist editor"
+            },
+            {
+                "key": "[shift] [Del]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kmplayer.json]: Let's capitalize "Shift" and any other key names to match "Ctrl" and "Del", etc

File: share/goodie/cheat_sheets/json/kmplayer.json (hunk 23 above)
Comment (mesooryanarayananb) [inline share/goodie/cheat_sheets/json/kmplayer.json]: i dont have a source url , is it okay not to add it ?? @moollaza

**Comment by mesooryanarayananb** [comment]
@moollaza all good now ??? i dont have any source url ?? is it necessary to add it ??

**Comment by moollaza** [comment]
>i dont have any source url ?? is it necessary to add it ?? 

Yes. You should provide a link to the source used, or at the very least so the user can click through to see more details.

This is basically done but still needs some cleanup. To make things quick, I'll finish it for you :+1: 

Thanks for taking the time to contribute! We really appreciate it.

We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by mesooryanarayananb** [comment]
@moollaza  thq :)  , I didn't copied it from an source , I used kmplayer itself !! ehh ??? so ???

**Comment by gautamkrishnar** [comment]
@mesooryanarayananb Congrats man! looking foreword for more pull requests from you..
:clap:  :smile:  :+1:

**Comment by mesooryanarayananb** [comment]
@gautamkrishnar  , aashane :D !! eeeeeee , ith innum epol aa site ille verunne ?

**Comment by gautamkrishnar** [comment]
@moollaza its better to remove the work in progress label from this pr as it is merged :smile:

**Comment by gautamkrishnar** [comment]
@mesooryanarayananb Please close this PR, It is now merged.
It will go live soon man!

**Comment by mintsoft** [comment]
@gautamkrishnar it's already closed; they are closed automatically by being merged

**Comment by gautamkrishnar** [comment]
@mintsoft Oh sorry i forgot :smile:

**Comment by mesooryanarayananb** [comment]
@gautamkrishnar eeee !!!  hurrray !!!

**Comment by moollaza** [comment]
@mesooryanarayananb congrats! It's live: https://duckduckgo.com/?q=kmplayer+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by gautamkrishnar** [comment]
Congrats man! :smile:

**Pull request merged**